homeassistant/components/simplefin/sensor.py

```diff
@@ -0,0 +1,127 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any
+
+from simplefin4py import Account
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SimpleFinDataUpdateCoordinator
+from .entity import SimpleFinEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class SimpleFinSensorEntityDescription(SensorEntityDescription):
+    """Describes a sensor entity."""
+
+    value_fn: Callable[[Account], int | str | datetime | None]
+
+    icon_fn: Callable[[Account], str] | None = None
+    unit_fn: Callable[[Account], str] | None = None
+
+    extra_state_attributes_fn: Callable[[Account], dict[str, Any]] | None = None
+
+
+SIMPLEFIN_SENSORS: tuple[SimpleFinSensorEntityDescription, ...] = (
+    SimpleFinSensorEntityDescription(
+        key="balance",
+        translation_key="balance",
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.MONETARY,
+        value_fn=lambda account: account.balance,
+        unit_fn=lambda account: account.currency,
+        icon_fn=lambda account: account.inferred_account_type,
```

Comment:
What's the type of `account.inferred_account_type`?

Reply:
```
class AccountType(Enum):
    """Account type enum."""

    CHECKING = "mdi:checkbook"
    CREDIT_CARD = "mdi:credit-card"
    SAVINGS = "mdi:piggy-bank-outline"
    INVESTMENT = "mdi:chart-areaspline"
    MORTGAGE = "mdi:home-city-outline"
    UNKNOWN = "mdi:cash"
```